torchtext/experimental/datasets/language_modeling.py

```diff
@@ -1,237 +0,0 @@
-                                                                year=year, language=language)
-        else:
-            raw_train, = raw.DATASETS[dataset_name](root=root, split=('train',))
-        logger_.info('Building Vocab based on train data')
-        vocab = build_vocab(raw_train, tokenizer)
-    logger_.info('Vocab has %d entries', len(vocab))
-
-    def text_transform(line):
-        return torch.tensor([vocab[token] for token in tokenizer(line)], dtype=torch.long)
-
-    if dataset_name == 'WMTNewsCrawl':
-        raw_datasets = experimental_raw.DATASETS[dataset_name](root=root, split=split,
-                                                               year=year, language=language)
-    else:
-        raw_datasets = raw.DATASETS[dataset_name](root=root, split=split)
-    raw_data = {name: list(map(text_transform, raw_dataset)) for name, raw_dataset in zip(split, raw_datasets)}
-    logger_.info('Building datasets for {}'.format(split))
-    return _wrap_datasets(tuple(LanguageModelingDataset(raw_data[item], vocab, text_transform)
-                                for item in split), split_)
-
-
-def WikiText2(tokenizer=None, root='.data', vocab=None, split=('train', 'valid', 'test')):
-    """ Defines WikiText2 datasets.
-
-    Create language modeling dataset: WikiText2
-    Separately returns the train/test/valid set
-
-    Args:
-        tokenizer: the tokenizer used to preprocess raw text data.
-            The default one is basic_english tokenizer in fastText. spacy tokenizer
-            is supported as well (see example below). A custom tokenizer is callable
-            function with input of a string and output of a token list.
-        root: Directory where the datasets are saved. Default: ".data"
-        vocab: Vocabulary used for dataset. If None, it will generate a new
-            vocabulary based on the train data set.
-        split: a string or tuple for the returned datasets. Default: ('train', 'valid','test')
-            By default, all the three datasets (train, test, valid) are generated. Users
-            could also choose any one or two of them, for example ('train', 'test') or
-            just a string 'train'. If 'train' is not in the tuple or string, a vocab
-            object should be provided which will be used to process valid and/or test
-            data.
-
-    Examples:
-        >>> from torchtext.experimental.datasets import WikiText2
-        >>> from torchtext.data.utils import get_tokenizer
-        >>> tokenizer = get_tokenizer("spacy")
-        >>> train_dataset, valid_dataset, test_dataset = WikiText2(tokenizer=tokenizer)
-        >>> vocab = train_dataset.get_vocab()
-        >>> valid_dataset = WikiText2(tokenizer=tokenizer, vocab=vocab,
-                                       split='valid')
-
-    """
-    return _setup_datasets("WikiText2", tokenizer, root, vocab, split, None, None)
-
-
-def WikiText103(tokenizer=None, root='.data', vocab=None, split=('train', 'valid', 'test')):
```

Comment:
Since we removed this, we also need to remove a dependent test here https://github.com/pytorch/text/blob/57a1df3ab450934c192e300a2e2e882e3e94a290/test/experimental/test_with_asset.py#L42 (we may completely remove the TestWithAsset class as it contains this only test)

Reply:
We may also remove corresponding asset https://github.com/pytorch/text/blob/master/test/asset/wikitext103_vocab.pt